src/assets/specs/command-api.yaml

```diff
@@ -112,6 +112,58 @@ paths:
             "text/plain":
               schema:
                 type: string
+
+  /v0/diagnostic_categories:
+    get:
+      operationId: diagnosticCategories
+      summary: >-
+        Return a list of the category names for the Diagnostics component. Takes no parameters
+      responses:
+        '200':
+          description: >-
+            A list of the category names
+          content:
+            "application/json":
+              schema:
+                type: array
+
+  /v0/diagnostic_ids:
+    get:
+      operationId: diagnosticIDsForCategory
+      summary: >-
+        Return a list of the check IDs for the Diagnostics category, or 404 if there is no such `category`. Specifying an
+        exiting category with no checks will return status code 200 and an empty array.
+      parameters:
+        - in: query
+          name: category
+      responses:
+        '200':
+          description: >-
+            A list of the check IDs for the specified category, or 404.
+          content:
+            "application/json":
+              schema:
+                type: array
+
+  /v0/diagnostic_check:
+    get:
+      operationId: diagnosticForCategoryAndID
+      summary: >-
+        Return the check for the given category and checkID, or 404.
+      parameters:
+        - in: query
+          name: category
```

Comment:
Is the category necessary? Presumably the id would be unique across all categories?

Reply:
It probably can be dropped, and I'd have to verify that all IDs are unique, but for now I still want the hierarchy (for ease of use from the UI, as explained above).